### DISCUSSION TITLE
Select the number of variables

### DISCUSSION THREAD

**Discussion by NicolasL-S** [JuliaSmoothOptimizers/CUTEst.jl#247] [Q&A] [answered]
Hi, I've been using CUTEst.jl for some weeks, it works great. But I have one issue. Most CUTEst problems have a variable number of variables. But after searching everywhere, I can't figure out whether we can change it when calling CUTEstModel(). Is it possible? Thanks for the great work.

**Comment by abelsiqueira** [comment, marked as answer]
Hi @NicolasL-S, thanks for using CUTEst.jl. It should be possible to specify the values of parameters by passing additional arguments, e.g.,
```julia
nlp = CUTEstModel("DIXMAANJ", "-param", "M=30")
```
These arguments are passed to the `sifdecoder` call. I do believe you will still need to verify the parameter and how it affects the number of variables looking into the .SIF file, but @dpo might correct me.

**Comment by dpo** [comment]
It's a bit [hidden in the documentation](https://juliasmoothoptimizers.github.io/CUTEst.jl/stable/api/#CUTEst.sifdecoder-Tuple{AbstractString,Vararg{Any,N}%20where%20N}).

Asking for help at the Julia prompt
```julia
?help sifdecoder
```
should also reveal the same example.

Admittedly, it should be more visible.

You may be able to call `CUTEst.sifdecoder("DIXMAANJ", "-show")` to see a list of allowed parameter values (I haven't checked). In general, as @abelsiqueira said, you still have to look in the SIF file. That should definitely be improved.

**Comment by NicolasL-S** [comment]
Thanks a lot for the two very quick replies which perfectly solved my problem. As always, great support from the welcoming Julia community.
